exchange/cache.go

```diff
@@ -140,7 +140,13 @@ func GetResourceFromCache(resourceKey string, resourceType string, expirationS u
 	if expirationS > 0 && expired {
 		return nil
 	}
-	return typedEntry.Resource
+	var entryResource interface{}
+	entryResource = typedEntry.Resource
+	if reflect.ValueOf(typedEntry.Resource).Type().Kind() == reflect.Ptr {
+		tempCopy := reflect.Indirect(reflect.ValueOf(typedEntry.Resource))
```

Comment:
This isnt making a copy of the map in the Resource. The problem is still recreate-able with this fix on.